test/allowlist_for_publicAPI.json

```diff
@@ -550,6 +550,7 @@
     "ForkingPickler",
     "Union",
     "check_serializing_named_tensor",
+    "register",
```

Comment:
Adding new entries to this list is not acceptable.
Please make sure to properly expose any API being changed or added.

Reply:
Looking at assertion in `test_public_bindins.py`, the other way I could've done this is to import the module and qualify the fn call with the module name:

```
from multiprocessing import reduction

reduction.register(...)
```

changed to using ^^^ since its cleaner than 

```
from multiprocessing.reduction import register as _register

_register(...)
```

Reply:
Well, it does work if it comes from a submodule: torch.nn.Linear is ok coming from torch.nn.modules.linear.Linear.

But in this case, you have torch.multiprocessing.reduction.reduce come from torch.multiprocessing.reductions.reduce (which is not a submodule).

Using reduction.register() sounds like a good way to go here for this special case.